src/app/src/main/java/com/example/bookmark/util/FormValidator.java

```diff
@@ -0,0 +1,68 @@
+package com.example.bookmark.util;
+
+import android.widget.EditText;
+
+import com.google.android.material.textfield.TextInputLayout;
+
+public class FormValidator {
+    /**
+     * Function to check if a EditText field is empty and float an error if it is.
+     *
+     * @args EditText editText: the text field to check
+     * boolean hasFocus: whether or not the text field has focus
+     * @returns boolean: whether or not the field is empty
+     */
+    public static boolean validateEditTextEmpty(EditText editText, TextInputLayout layout) {
+        if (editText.getText().toString().equals("")) {
+            layout.setError("This field cannot be left empty.");
+            return false;
+        } else {
+            layout.setError(null);
+            return true;
+        }
+    }
+
+    /**
+     * Function to check if a EditText field is a valid email and float an error if it is
+     *
+     * @args EditText editText: the text field to check
+     * boolean hasFocus: whether or not the text field has focus
+     * @returns boolean: whether or not the editText contains a valid email
+     */
+    public static boolean checkIfEditTextValidEmail(EditText editText, TextInputLayout layout) {
+        String emailPattern = "[a-zA-Z0-9._-]+@[a-z]+\\.+[a-z]+";
+        String string = editText.getText().toString();
+        if (string.equals("")) {
+            layout.setError("This field cannot be left empty");
+            return false;
+        } else if (!string.matches(emailPattern)) {
+            layout.setError("Please enter a valid email address.");
+            return false;
+        } else {
+            layout.setError(null);
+            return true;
+        }
+    }
+
+    /**
+     * Function to check if a EditText field is a valid phone number and float an error if it is
+     *
+     * @args EditText editText: the text field to check
+     * boolean hasFocus: whether or not the text field has focus
+     * @returns boolean: whether or not the editText contains a phone number
+     */
+    public static boolean checkIfEditTextValidPhoneNumber(EditText editText, TextInputLayout layout) {
+        String phonePattern = "^\\s*(?:\\+?(\\d{1,3}))?[-. (]*(\\d{3})[-. )]*(\\d{3})[-. ]*(\\d{4})(?: *x(\\d+))?\\s*$";
```

Comment:
... I'm just gonna trust you on this one lol

Reply:
yep...

Reply:
> Code looks good, I fired it up in emulator too and visuals look great. Something I might consider for an extra polish (though this isn't a request for it to be in this PR) is to do some automatic formatting of the phone number on focus loss (automatically adding the hyphens, maybe the (###) ###-#### if you are feeling fancy). I did something similar for the gear project and it wasn't too hard. Feel free to ask me about it if curious.
> 
> I'll approve after a couple of my comments are resolved!

I like the idea of automatic formatting of phone number on formatting loss. I'll add that as a personal stretch goal. The current phone number validation that I am using is for basically any phone number on earth, which makes it somewhat difficult to guess the exact format. I may change the regex to limit it slightly for when i do that.